crates/subspace-farmer-components/src/sector.rs

```diff
@@ -92,25 +92,31 @@ impl SectorMetadata {
 
 /// Commitment and witness corresponding to the same record
 #[derive(Debug, Default, Clone, Encode, Decode)]
-pub struct RecordMetadata {
+pub(crate) struct RecordMetadata {
```

Comment:
Don't you think `pub(crate)` for the struct only is enough? Is it somewhere in the rust guidelines?

Reply:
Fields and methods were also made private, what do you mean by struct only?

I was just hiding data structures that are internal to `subspace-farmer-components` and not exposed in public API. This helps ensuring that when in later PR I'm adding checksums, all users of these data structures are doing checksums the same way consistently.

Reply:
I meant that by making the struct private you prevent leaking the type outside of the crate and all methods and fields are unaccessible respectively. `pub` methods and fields are less noisy than `pub(crate)`. But maybe it's considered malpractice by the Rust guideline with a good reason.

Reply:
I mean if you need to make it visible publicly, you use `pub`, if only in parent module you use `pub(super)` and if only inside of the crate then `pub(crate)`. Using `pub` when it is not exposed publicly is misleading and also results in data structures not reported as unused after refactoring. So everything that can be private should be private and as private as possible.